File: modules/opla-front/src/app/mlconfig/mlmodels/RandomTreeLearningModel.ts
import {MachineLearningModel} from "./MachineLearningModel";
import {MachineLearningAlgorithm} from "./MachineLearningAlgorithm";

export class RandomTreeLearningModel extends MachineLearningModel{

  kValue: string;
  breakTiesRandomly: boolean;
  maxDepth: string;
  minNum: string;
  minVariancePop: string;
  numFolds: string;


  constructor(kValue: string, breakTiesRandomly: string, maxDepth: string, minNum: string, minVariancePop: string, numFolds: string) {
    super(MachineLearningAlgorithm.RANDOM_TREE);
    this.kValue = kValue;
    this.breakTiesRandomly = (breakTiesRandomly === "true");
    this.maxDepth = maxDepth;
    this.minNum = minNum;
    this.minVariancePop = minVariancePop;
    this.numFolds = numFolds;
  }
}

File: modules/opla-core/src/main/java/br/otimizes/oplatool/core/learning/mlmodels/RandomTreeLearningModel.java
package br.otimizes.oplatool.core.learning.mlmodels;

public class RandomTreeLearningModel {
    private int kValue;
    private boolean breakTiesRandomly;
    private int maxDepth;
    private double minNum;
    private double minVariancePop;
    private int numFolds;

    public RandomTreeLearningModel(int kValue, boolean breakTiesRandomly, int maxDepth, double minNum, double minVariancePop, int numFolds) {
        this.kValue = kValue;
        this.breakTiesRandomly = breakTiesRandomly;
        this.maxDepth = maxDepth;
        this.minNum = minNum;
        this.minVariancePop = minVariancePop;
        this.numFolds = numFolds;
    }

    public int getkValue() {
        return kValue;
    }

    public void setkValue(int kValue) {
        this.kValue = kValue;
    }

    public boolean isBreakTiesRandomly() {
        return breakTiesRandomly;
    }

    public void setBreakTiesRandomly(boolean breakTiesRandomly) {
        this.breakTiesRandomly = breakTiesRandomly;
    }

    public int getMaxDepth() {
        return maxDepth;
    }

    public void setMaxDepth(int maxDepth) {
        this.maxDepth = maxDepth;
    }

    public double getMinNum() {
        return minNum;
    }

    public void setMinNum(double minNum) {
        this.minNum = minNum;
    }

    public double getMinVariancePop() {
        return minVariancePop;
    }

    public void setMinVariancePop(double minVariancePop) {
        this.minVariancePop = minVariancePop;
    }

    public int getNumFolds() {
        return numFolds;
    }

    public void setNumFolds(int numFolds) {
        this.numFolds = numFolds;
    }
}
